### PR TITLE
feat: Initialize only containerizers of interest

### PR DESCRIPTION
**Pull request by ashokponkumar**
Signed-off-by: Ashok Pon Kumar <ashokponkumar@gmail.com>

### CODE REVIEW

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
Some doubts

File: internal/move2kube/planner.go [hunk 1]
```diff
@@ -27,12 +28,31 @@ import (
...
+func CreatePlan(inputPath string, prjName string, interactive bool) plantypes.Plan {
 	p := plantypes.NewPlan()
 	p.Name = prjName
 	p.Spec.Inputs.RootDir = inputPath
 
-	translationPlanners := source.GetSourceLoaders()
+	translationPlanners := source.GetTranslators()
+	if interactive {
+		translationTypes := selectTranslators(source.GetAllTranslatorTypes())
+		translationPlanners := []source.Translator{}
```
[the remark this answers is not in the document]
Comment (ashokponkumar) [reply internal/move2kube/planner.go:39]: Good find.

File: internal/move2kube/planner.go [hunk 2]
```diff
@@ -81,18 +101,8 @@ func CuratePlan(p plantypes.Plan) plantypes.Plan {
...
-	}
-	problem, err = qaengine.FetchAnswer(problem)
-	if err != nil {
-		log.Fatalf("Unable to fetch answer : %s", err)
-	}
-	selectedTranslationTypes, err := problem.GetSliceAnswer()
-	if err != nil {
-		log.Fatalf("Unable to get answer : %s", err)
-	}
+	selectedTranslationTypes := selectTranslators(translationTypes)
```
[the remark this answers is not in the document]
Comment (ashokponkumar) [reply internal/move2kube/planner.go:104]: yes, that's why we use the cache to answer automatically if the same question is required in two places.

File: internal/move2kube/planner.go [hunk 3]
```diff
@@ -274,3 +277,38 @@ func CuratePlan(p plantypes.Plan) plantypes.Plan {
...
+	problem, err = qaengine.FetchAnswer(problem)
+	if err != nil {
+		log.Fatalf("Unable to fetch answer : %s", err)
+	}
+	selectedConTypes, err = problem.GetSliceAnswer()
+	if err != nil {
+		log.Fatalf("Unable to get answer : %s", err)
+	}
+	if len(selectedConTypes) == 0 {
+		log.Fatalf("No containerization technique was selected; Terminating.")
```
[the remark this answers is not in the document]
Comment (ashokponkumar) [reply internal/move2kube/planner.go:311]: Yep.

File: internal/move2kube/translator.go [hunk 4]
```diff
@@ -32,6 +34,13 @@ import (
 
 // Translate translates the artifacts and writes output
 func Translate(p plantypes.Plan, outpath string, qadisablecli bool) {
+	conTypes := []string{}
+	for _, s := range p.Spec.Inputs.Services {
+		if !common.IsStringPresent(conTypes, string(s[0].ContainerBuildType)) {
```
[the remark this answers is not in the document]
Comment (ashokponkumar) [reply internal/move2kube/translator.go:39]: It should not happen, but yes, good to have a check.

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**